web/static/js/components/import_export_tray.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react'
+import CopyToClipboard from 'react-copy-to-clipboard'
+
+class ImportExportTray extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      pageContents: this.props.pageContents,
+      pageSettings: this.props.pageSettings,
+      stringifiedExportData: '',
+      stringifiedImportData: '',
+      exportDataButtonText: 'Copy to Clipboard'
+    }
+
+    this.onExportCopy = this.onExportCopy.bind(this)
+    this.stringifyExportData = this.stringifyExportData.bind(this)
+    this.onImportDataPaste = this.onImportDataPaste.bind(this)
+    this.onImportApply = this.onImportApply.bind(this)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({
+      pageContents: nextProps.pageContents,
+      pageSettings: nextProps.pageSettings,
+      stringifiedExportData: '',
+      stringifiedImportData: '',
+      exportDataButtonText: 'Copy to Clipboard'
+    })
+  }
+
+  componentDidMount () {
+    this.stringifyExportData()
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (this.state.stringifiedExportData !== '') return
+
+    this.stringifyExportData()
+  }
+
+  onExportCopy () {
+    this.setState({exportDataButtonText: 'Copied!'})
+
+    clearTimeout(window.exportDataButtonTextTimeout)
+    window.exportDataButtonTextTimeout = setTimeout(() => {
+      this.setState({exportDataButtonText: 'Copy to Clipboard'})
+    }, 2000)
+  }
+
+  stringifyExportData () {
+    let exportData = {
+      pageSettings: Object.assign({}, this.state.pageSettings, {origin: window.location.origin}) ,
+      pageContents: this.state.pageContents
+    }
+
+    const stringifiedExportData = JSON.stringify(exportData)
+
+    this.setState({
+      stringifiedExportData: stringifiedExportData
+    })
+  }
```

Comment:
this create a json string from the data. we have the page settings object, and the contents array. also, we can add more useful data into the page settings that will be needed during import